pkg/upgrade/job/job.go

```diff
@@ -137,16 +137,40 @@ func New(plan *upgradeapiv1.Plan, node *corev1.Node, controllerName string) *bat
 						NodeAffinity: &corev1.NodeAffinity{
 							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
 								NodeSelectorTerms: []corev1.NodeSelectorTerm{{
-									MatchExpressions: []corev1.NodeSelectorRequirement{{
-										Key:      corev1.LabelHostname,
-										Operator: corev1.NodeSelectorOpIn,
-										Values: []string{
-											nodeHostname,
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      corev1.LabelHostname,
+											Operator: corev1.NodeSelectorOpIn,
+											Values: []string{
+												nodeHostname,
+											},
 										},
-									}},
+										{
+											Key:      corev1.LabelOSStable,
+											Operator: corev1.NodeSelectorOpIn,
+											Values: []string{
+												"linux",
+											},
+										},
+									},
 								}},
 							},
+							PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+								{
+									Weight: 1,
+									Preference: corev1.NodeSelectorTerm{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      "node-role.kubernetes.io",
+												Operator: corev1.NodeSelectorOpIn,
+												Values: []string{
+													"control-plane",
+												},
+											}},
+									},
+								}},
```

Comment:
Is the idea here to favor control-plane nodes for upgrade prior to all other nodes? While that would be an valuable enhancement to what SUC can currently do, please be aware that SUC generates one job per node to which a plan's node-selector applies. Jobs are only ever destined to land on a singular node and so this scheduling preference should never come into play (the node for the generated job has already been picked).

Furthermore, this changes seems superfluous in that, as I understand it, rancher/rancher implements its upgrade controller (that leverages SUC) in such a way that relies on separate, inter-locking plans: one for "servers" and one for "workers" (via node-selector).